### PR TITLE
feat: create new event module for decouple logic and specific library

### PR DESCRIPTION
**Pull request by bzp2010**
### Description

Since we wanted to introduce a new implementation (#10550) of the event library, and the current code strongly relies on `lua-resty-worker-events`.

Therefore we need a new abstraction layer to accommodate the library API differences and to unify the usage logic within the APISIX gateway. So that the code no longer connects directly to a specific library.

This PR is supposed to not break any existing tests, as there are no substantive changes at this time.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by soulbird** [comment]
> Should the `apisix.event` module use the plural form `apisix.events`? @monkeyDluffy6017 @soulbird

`apisix.events` will be better

**Comment by monkeyDluffy6017** [comment]
@bzp2010 i prefer to `apisix.events`

File: apisix/event.lua [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    local ok, err = we.configure({shm = shm, interval = 0.1})
+    if not ok then
+        error("failed to init worker event: " .. err)
+    end
+
+    return we
+end
+
+
+function _M.init_worker()
```
Comment (monkeyDluffy6017) [inline apisix/event.lua:41]: should we pass the option?

Comment (bzp2010) [reply apisix/event.lua:41]: @monkeyDluffy6017 Do you mean some options for the events module, such as the configuration from yaml?

Comment (monkeyDluffy6017) [reply apisix/event.lua:41]: Yeah,  it's better if we can do that in the `config.yaml`

Comment (bzp2010) [reply apisix/event.lua]: It looks like we don't need any configuration for existing use cases. Perhaps we'll need to configure that later when we provide a switchable events module implementation.

Comment (bzp2010) [reply apisix/event.lua]: @monkeyDluffy6017 To summarize, we may introduce some options in future PR, if you have concerns about the continuity of implementation, we can create a separate new branch for staging each PR and commit the integrated code to the mainline from the new branch.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by AlinsRan — APPROVED** [review]

**Pull request merged**